developer/wip/for-contributors/data-plane/custom-data-plane.md

```diff
@@ -0,0 +1,252 @@
+    });
+
+    const { sourceDataAddress, destinationDataAddress } = req.body;
+
+    const sourceUrl = getBaseUrl(sourceDataAddress);
+    const destinationUrl = getBaseUrl(destinationDataAddress);
+
+    const response = await fetch(sourceUrl);
+
+    await fetch(destinationUrl, {
+        "method": "POST",
+        body : await response.text()
+    });
+}
+```
+
+First we acknowledge the [Control Plane](../contributor-handbook.md#2-the-control-plane) by sending a `DataFlowResponseMessage` as response. 
+
+Then we transfer the data from `sourceUrl` to `destinationUrl`.
+
+> The `getBaseUrl` is an utility function that extract the `baseUrl` from the `DataAddress`.
+ 
+Implementors should keep track of `DataFlowStartMessage`s in some persistent storage system in order to fulfill subsequent `DPS` messages on the same transfer id ([e.g. SUSPEND and TERMINATE](#22-suspend-and-terminate)).
+
+For example in the streaming case, implementors may track the opened streaming channels, which could be terminated on-demand or by the [policy monitor](../control-plane/policy-monitor.md).
+
+
+### 2.1.2 `PULL`
+
+When receiving a `DataFlowStartMessage` in a `PULL` scenario there is no direct transfer to be handled by the data plane. Based on the `sourceDataAddress` in the `DataFlowStartMessage` a custom data plane implementation should create another `DataAddress` containing all the information required for the data transfer:
+
+
+```javascript
+async function handlePull(req, res) {
+    const { sourceDataAddress } = req.body;
+    const { dataAddress } = await generateDataAddress(sourceDataAddress);
+
+    const response = {
+        "@context": {
+            "edc": "https://w3id.org/edc/v0.0.1/ns/"
+        },
+        "@type": "DataFlowResponseMessage",
+        "dataAddress": dataAddress
+    };
+    res.send(response);
+}
+```
+
+We will not implement the `generateDataAddress` function, as it may vary depending on the use case. But at high level a `generateDataAddress` should generate a `DataAddress` in DSP format that contains useful information for the consumer for fetching the data: `endpoint`, `endpointType` and custom extensible properties `endpointProperties`. 
+
+For example the default [EDC](./data-plane-signaling/data-plane-signaling.md#323-access-token-generation) genarates a `DataAddress` that contains also authorization information like the auth token to be used when request data using the Data Plane [public API](./data-plane-signaling/data-plane-signaling.md#3-data-plane-public-api) and the token type (e.g. bearer).
+
+Implementors may also want to track `PULL` requests in a persistent storage, which can be useful in scenario like token revocation or transfer process termination.
+
+How the actual data requests is handled depends on the implementation of the custom data plane. It could be like the EDC default one, where it exposes an endpoint that validates the authorization and it proxies the request to the `sourceDataAddress`. 
```

Comment:
```suggestion
How the actual data requests is handled depends on the implementation of the custom data plane. It could be done in the same way as it's done in the EDC data plane, which exposes an endpoint that validates the authorization and it proxies the request to the `sourceDataAddress`. 
```